src/System.Numerics.Tensors/System/Numerics/CompressedSparseTensor.cs

```diff
@@ -202,14 +205,23 @@ private void EnsureCapacity(int min, int allocateIndex = -1)
                         // copy range before allocateIndex
                         if (allocateIndex > 0)
                         {
-                            Array.Copy(values, newValues, allocateIndex);
-                            Array.Copy(indices, newIndices, allocateIndex);
+                            var valuesSpan = values.Span.Slice(0, allocateIndex);
+                            var indicesSpan = indices.Span.Slice(0, allocateIndex);
+
+                            valuesSpan.CopyTo(newValues.Span);
+                            indicesSpan.CopyTo(newIndices.Span);
                         }
 
                         if (allocateIndex < nonZeroCount)
                         {
-                            Array.Copy(values, allocateIndex, newValues, allocateIndex + 1, nonZeroCount - allocateIndex);
-                            Array.Copy(indices, allocateIndex, newIndices, allocateIndex + 1, nonZeroCount - allocateIndex);
+                            var valuesSpan = values.Span.Slice(allocateIndex, nonZeroCount - allocateIndex);
+                            var indicesSpan = indices.Span.Slice(allocateIndex, nonZeroCount - allocateIndex);
+
+                            var newValuesSpan = newValues.Span.Slice(allocateIndex + 1, nonZeroCount - allocateIndex);
+                            var newIndicesSpan = newIndices.Span.Slice(allocateIndex + 1, nonZeroCount - allocateIndex);
+
+                            valuesSpan.CopyTo(newValuesSpan);
```

Comment:
This felt like a lot of ceremony around just shifting bytes to the right.  Potential opportunity for API improvements on Span or Memory here.

One thing that comes to mind: a method to slice a span directly from memory, instead of accessing a span and slicing the span.